src/main/plugin-metadata/plugin-security.policy

```diff
@@ -6,6 +6,10 @@
  * compatible open source license.
  */
 
+grant {
+  permission java.net.NetPermission "accessUnixDomainSocket";
```

Comment:
Why is this a new permission? Are we actually using unix domain sockets in our code? And why was it not previously restricted by the security manager?

Reply:
Yes .... On Windows - UNIX domain sockets

Reply:
> And why was it not previously restricted by the security manager?

I think because we now [intercept](https://github.com/opensearch-project/OpenSearch/blob/967eee15eeb4219f2e7e5275c132cf86deff226a/libs/agent-sm/agent/src/main/java/org/opensearch/javaagent/SocketChannelInterceptor.java#L64) it explicitly.

Reply:
If this permission has to go in every plugin, do we need to enforce the [interception](https://github.com/opensearch-project/OpenSearch/blob/967eee15eeb4219f2e7e5275c132cf86deff226a/libs/agent-sm/agent/src/main/java/org/opensearch/javaagent/SocketChannelInterceptor.java#L64) on Unix domain Sockets?   I think that we should trust IPC calls. Anything which is inbound / outbound should be intercepted.

Reply:
> If this permission has to go in every plugin, do we need to enforce the [interception](https://github.com/opensearch-project/OpenSearch/blob/967eee15eeb4219f2e7e5275c132cf86deff226a/libs/agent-sm/agent/src/main/java/org/opensearch/javaagent/SocketChannelInterceptor.java#L64) on Unix domain Sockets? I think that we should trust IPC calls. Anything which is inbound / outbound should be intercepted.

Intuitively, there could be anything on the other IPC side, so it does look safer to explicitly grant this permission, and it does not look to be much of a burden to add one I think

Reply:
> so it does look safer to explicitly grant this permission, and it does not look to be much of a burden to add one I think

@reta It seems like most or all plugins will need an update here. Is this new enforcement that security manager did not do?

Reply:
> @reta It seems like most or all plugins will need an update here. Is this new enforcement that security manager did not do?

@andrross not necessarily didn't do, but I suspect it was done differently.